rasa/core/actions/action.py

```diff
@@ -1153,6 +1154,12 @@ async def run(
         ]
 
         for slot in user_slots:
+            if (
+                tracker.active_loop
+                and tracker.active_loop.get(LOOP_REJECTED)
+                and tracker.get_slot(REQUESTED_SLOT) == slot.name
+            ):
+                continue
```

Comment:
Looks to me like this check belongs to `_matches_mapping_conditions`. Does that make sense or would we skip some case this way?

Reply:
i think it would clutter `_matches_mapping_conditions` with a 2nd use-case that has more to do with the tracker state than with what was defined as mapping conditions in the domain. See also my comment [here](https://github.com/RasaHQ/rasa/pull/10295/files#r756768236).

Reply:
@ka-bu Do you know though why `LoopInterrupted` event would not get added immediately after the user utters `stop`? It's been bugging me for a few days now 😄

Reply:
Hmmm, in my brain it would unclutter things if this was part of checking whether the loop conditions is *really* met 😅  (Couldn't we have un-intended side effects from skipping all slot mappings? 🤔 )

>  why LoopInterrupted event would not get added immediately after the user utters stop?

Think that's because the "/<command>" stuff first becomes a special intent and then the rule policy takes care of inserting the `LoopInterrupted` and that next action prediction stuff happens as last step. Does that make sense?

Reply:
> Couldn't we have un-intended side effects from skipping all slot mappings?

Oh you're right! We'll aim to move this into `_matches_mapping_conditions` 👍🏼 

> the "/" stuff first becomes a special intent 

This is the part I don't understand how it happens 🙈 

> the rule policy takes care of inserting the LoopInterrupted

I should have said `LoopInterrupted` is added only after the 2nd user uttered in this sequence, so it wasn't clear to me why it wouldn't get added before `utter_ask_continue` 🤔 :
`stop` -> `utter_ask_continue`-> `affirm` -> `SlotSet` -> `LoopInterrupted`

Reply:
>  is added only after the 2nd user uttered in this sequence

ahhh, ok, that looks weird - will try to find out 🕵️‍♀️

Reply:
Yeah if the `LoopInterrupted` would be added before the `utter_ask_continue`, it would have simplified our check in the `tracker` for the unhappy path, rather than trying to access the `LOOP_REJECTED` in `tracker.active_loop` which felt a bit like mental gymnastics 🤪

Reply:
Also, just read your comment on `RegexMessageHandler` for handling special intents, going to study that a bit 👍🏼

Reply:
Sooo, seems like...
- a rule policy will only cause the `LoopInterrupted` to be added if it makes a prediction where `returning_from_unhappy_path = True` but that is not set when it predicts a (i.e. any) default action ( https://github.com/RasaHQ/rasa/blob/main/rasa/core/policies/rule_policy.py#L1123) --> so the stop on it's own has no implications / there is no "default rule" being added for any intent
- but during training, the augmented memoization policy should learn from the stories which includes this:
```
  - story: stop form + stop
    steps:
      - intent: request_restaurant
      - action: restaurant_form
      - active_loop: restaurant_form
      - intent: stop. <-- **stop here**
      - action: utter_ask_continue
      - intent: deny
      - action: action_deactivate_loop
      - active_loop: null <-- **deactivate much later than we think**
```
Does that make sense? 🤔

Reply:
I see, so we stick with the current check for `LOOP_REJECTED`

Reply:
doh, it doesn't make sense because we affirm there instead of denying ... wishful thinking 😅